prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -50,66 +50,80 @@ class SendFunction(private val workflowEngine: WorkflowEngine = WorkflowEngine()
         @BindingName("NextVisibleTime") nextVisibleTime: Date? = null,
         @BindingName("InsertionTime") insertionTime: Date? = null,
     ) {
-        val event = Event.parseQueueMessage(message) as ReportEvent
-        val actionHistory = ActionHistory(TaskAction.send, context, event.isEmptyBatch)
-        actionHistory.trackActionParams(message)
-        context.logger.info(
-            "Started Send Function: $message, id=$messageId," +
-                " dequeueCount=$dequeueCount, " +
-                " nextVisibleTime=$nextVisibleTime, insertionTime=$insertionTime"
-        )
         try {
-            if (event.eventAction != Event.EventAction.SEND) {
-                context.logger.warning("Send function received a $message")
-                return
-            }
-            workflowEngine.handleReportEvent(event, context) { header, retryToken, _ ->
-                val receiver = header.receiver
-                    ?: error("Internal Error: could not find ${header.task.receiverName}")
-                val inputReportId = header.reportFile.reportId
-                actionHistory.trackExistingInputReport(inputReportId)
-                val serviceName = receiver.fullName
-                val nextRetryItems = mutableListOf<String>()
-                if (receiver.transport == null) {
-                    actionHistory.setActionType(TaskAction.send_warning)
-                    actionHistory.trackActionResult("Not sending $inputReportId to $serviceName: No transports defined")
-                } else {
-                    val retryItems = retryToken?.items
-                    val sentReportId = UUID.randomUUID() // each sent report gets its own UUID
-                    val nextRetry = getTransport(receiver.transport)?.send(
-                        receiver.transport,
-                        header,
-                        sentReportId,
-                        retryItems,
+            val event = Event.parseQueueMessage(message) as ReportEvent
+            val actionHistory = ActionHistory(TaskAction.send, context, event.isEmptyBatch)
+
```

Comment:
These two Vals were moved outside the try() and values that we needed to use and catch() created an error state that was solved most elegantly by using the external try().